diff --git a/arch/arm/mach-msm/Makefile b/arch/arm/mach-msm/Makefile
index a89843e..d153667 100644
--- a/arch/arm/mach-msm/Makefile
+++ b/arch/arm/mach-msm/Makefile
@@ -280,7 +280,7 @@ obj-$(CONFIG_MACH_MSM8930_EVT) += board-8930-all.o board-8930-regulator-pm8038.o
 obj-$(CONFIG_MACH_EXPRESS2) += board-express2.o board-8930-camera-power.o board-8930-camera.o board-8930-display.o board-8930-pmic.o board-8930-storage.o board-express2-gpiomux.o devices-8930.o board-8930-gpu.o board-8930-regulator-pm8038.o board-8930-regulator-pm8917.o board-8930-input-mxt.o board-8930-input-tkey.o
 obj-$(CONFIG_MACH_EXPRESS) += board-express.o board-8930-camera-power.o board-8930-camera.o board-8930-display.o board-8930-pmic.o board-8930-storage.o board-express-gpiomux.o devices-8930.o board-8930-gpu.o board-8930-regulator-pm8038.o board-8930-regulator-pm8917.o board-8930-input-mxt.o board-8930-input-tkey.o
 obj-$(CONFIG_MACH_HIGGS) += board-higgs.o board-8930-camera-power.o board-8930-camera.o board-8930-display.o board-8930-pmic.o board-8930-storage.o board-higgs-gpiomux.o devices-8930.o board-8930-gpu.o board-8930-regulator-pm8038.o board-8930-regulator-pm8917.o board-8930-input-mms.o
-obj-$(CONFIG_MACH_SERRANO) += board-serrano.o board-8930-camera-power.o board-8930-camera.o board-8930-display.o board-8930-pmic.o board-8930-storage.o board-serrano-gpiomux.o devices-8930.o board-8930-gpu.o board-8930-regulator-pm8038.o board-8930-regulator-pm8917.o board-8930-input-mxt.o board-8930-input-tkey.o
+obj-$(CONFIG_MACH_SERRANO) += board-serrano.o board-8930-camera-power.o board-8930-camera.o board-8930-display.o board-8930-pmic.o board-8930-storage.o board-serrano-gpiomux.o devices-8930.o board-8930-gpu.o board-8930-regulator-pm8038.o board-8930-regulator-pm8917.o board-8930-input-mxt.o board-8930-input-tkey-serrano.o
 obj-$(CONFIG_MACH_CRATER) += board-crater.o board-8930-camera-power.o board-8930-camera.o board-8930-display.o board-8930-pmic.o board-8930-storage.o board-crater-gpiomux.o devices-8930.o board-8930-gpu.o board-8930-regulator-pm8038.o board-8930-regulator-pm8917.o board-8930-input-mxt.o board-8930-input-tkey.o
 obj-$(CONFIG_MACH_BAFFIN) += board-baffin.o board-8930-camera-power.o board-8930-camera.o board-8930-display.o board-8930-pmic.o board-8930-storage.o board-baffin-gpiomux.o devices-8930.o board-8930-gpu.o board-8930-regulator-pm8038.o board-8930-regulator-pm8917.o board-8930-input-mxt.o board-8930-input-tkey.o
 obj-$(CONFIG_MACH_JASPER2) += board-jasper2.o board-8930-camera-power.o board-8930-camera.o board-8930-display.o board-8930-pmic.o board-8930-storage.o board-jasper2-gpiomux.o devices-8930.o board-8930-gpu.o board-8930-regulator-pm8038.o board-8930-regulator-pm8917.o
diff --git a/arch/arm/mach-msm/acpuclock-8930ab.c b/arch/arm/mach-msm/acpuclock-8930ab.c
index 63fd725..59f0e14 100644
--- a/arch/arm/mach-msm/acpuclock-8930ab.c
+++ b/arch/arm/mach-msm/acpuclock-8930ab.c
@@ -276,7 +276,7 @@ static struct acpu_level tbl_PVS6_1700MHz[] __initdata = {
 	{ 0, { 0 } }
 };
 
-#if defined(CONFIG_MACH_MELIUS_SKT)||defined(CONFIG_MACH_MELIUS_LGT)||defined(CONFIG_MACH_MELIUS_KTT)
+#if defined(CONFIG_MACH_MELIUS_SKT)||defined(CONFIG_MACH_MELIUS_LGT)||defined(CONFIG_MACH_MELIUS_KTT)||defined(CONFIG_MACH_SERRANO_SPR)
 static struct pvs_table pvs_tables[NUM_SPEED_BINS][NUM_PVS] __initdata = {
 	[0][0] = { tbl_PVS0_1700MHz, sizeof(tbl_PVS0_1700MHz), 25000 },
 	[0][1] = { tbl_PVS1_1700MHz, sizeof(tbl_PVS1_1700MHz), 50000 },
diff --git a/arch/arm/mach-msm/acpuclock-krait.c b/arch/arm/mach-msm/acpuclock-krait.c
index ecb8f01..02b8b06 100644
--- a/arch/arm/mach-msm/acpuclock-krait.c
+++ b/arch/arm/mach-msm/acpuclock-krait.c
@@ -434,7 +434,7 @@ module_param_named(boost, enable_boost, bool, S_IRUGO | S_IWUSR);
 
 static int calculate_vdd_core(const struct acpu_level *tgt)
 {
-#if defined(CONFIG_MACH_MELIUS_SKT)||defined(CONFIG_MACH_MELIUS_LGT)||defined(CONFIG_MACH_MELIUS_KTT)
+#if defined(CONFIG_MACH_MELIUS_SKT)||defined(CONFIG_MACH_MELIUS_LGT)||defined(CONFIG_MACH_MELIUS_KTT)||defined(CONFIG_MACH_SERRANO_SPR)
 	return tgt->vdd_core + drv.boost_uv ;
 #else
 	return tgt->vdd_core + (enable_boost ? drv.boost_uv : 0);
diff --git a/arch/arm/mach-msm/include/mach/serrano-gpio.h b/arch/arm/mach-msm/include/mach/serrano-gpio.h
index 4b52826..fbf6ae4 100644
--- a/arch/arm/mach-msm/include/mach/serrano-gpio.h
+++ b/arch/arm/mach-msm/include/mach/serrano-gpio.h
@@ -82,7 +82,7 @@
 #define GPIO_SPKR_I2S_RX_WS		62
 
 #if defined (CONFIG_MACH_SERRANO_ATT) || defined(CONFIG_MACH_SERRANO_VZW)\
-	|| defined(CONFIG_MACH_SERRANO_LRA)
+	|| defined(CONFIG_MACH_SERRANO_LRA) || defined(CONFIG_MACH_SERRANO_SPR)
 #define GPIO_VPS_AMP_EN		PM8038_GPIO_PM_TO_SYS(14)
 #else
 #define GPIO_VPS_AMP_EN		0
diff --git a/include/linux/i2c/cypress_touchkey.h b/include/linux/i2c/cypress_touchkey.h
index 549ea70..24b566c 100644
--- a/include/linux/i2c/cypress_touchkey.h
+++ b/include/linux/i2c/cypress_touchkey.h
@@ -14,7 +14,7 @@
 extern struct class *sec_class;
 extern int ISSP_main(void);
 extern int touch_is_pressed;
-extern int system_rev;
+extern unsigned int system_rev;
 
 #include <linux/i2c.h>
 #include <linux/delay.h>
