diff --git a/Android.mk b/Android.mk
index 51b44fa..98f7e74 100644
--- a/Android.mk
+++ b/Android.mk
@@ -24,7 +24,7 @@ LOCAL_PATH := $(call my-dir)
 
 ifeq ($(BOARD_VENDOR),samsung)
 ifeq ($(TARGET_BOARD_PLATFORM),msm8960)
-ifneq ($(filter lt02ltespr serrano3gxx serranodsdd serranodsub serranoltexx,$(TARGET_DEVICE)),)
+ifneq ($(filter lt02ltespr serrano3gxx serranodsdd serranodsub serranoltespr serranoltexx,$(TARGET_DEVICE)),)
 include $(call all-subdir-makefiles,$(LOCAL_PATH))
 endif
 endif
