diff --git a/Android.mk b/Android.mk
index cf134f3..3ae9e30 100644
--- a/Android.mk
+++ b/Android.mk
@@ -23,7 +23,7 @@ LOCAL_PATH := $(call my-dir)
 
 ifeq ($(BOARD_VENDOR),samsung)
 ifeq ($(TARGET_BOARD_PLATFORM),msm8960)
-ifneq ($(filter serrano3gxx serranodsdd serranodsub serranoltexx,$(TARGET_DEVICE)),)
+ifneq ($(filter serrano3gxx serranodsdd serranodsub serranoltespr serranoltexx,$(TARGET_DEVICE)),)
 include $(call all-subdir-makefiles,$(LOCAL_PATH))
 endif
 endif
diff --git a/setup-makefiles.sh b/setup-makefiles.sh
index 3553299..a795c5f 100755
--- a/setup-makefiles.sh
+++ b/setup-makefiles.sh
@@ -188,7 +188,7 @@ EOF
 
 LOCAL_PATH := \$(call my-dir)
 
-ifneq (\$(filter serrano3gxx serranodsub serranoltexx,\$(TARGET_DEVICE)),)
+ifneq (\$(filter serrano3gxx serranodsub serranoltespr serranoltexx,\$(TARGET_DEVICE)),)
 
 include \$(CLEAR_VARS)
 LOCAL_MODULE := libtime_genoff
